src/imitation_loss/imitation_loss.jl

```diff
@@ -0,0 +1,76 @@
+"""
+    ImitationLoss{L,R,P}
+
+Generic imitation loss: max_y ℓ(y, y̅) + α θᵀ(y - y̅) - (Ω(y) - Ω(y̅)).
+
+When `ℓ = 0`, this loss is equivalent to a [`FenchelYoungLoss`](@ref).
+When `Ω = 0`, this loss is equivalent to the [`StructuredSVMLoss`](@ref).
+
+# Fields
+- `maximizer::P`: function that computes
+    argmax_y ℓ(y, y̅) + α θᵀ(y - y̅) - (Ω(y) - Ω(y̅)), takes (θ, y_true, kwargs...)
+    or (θ, t_true, kwargs...) as input
+- `ℓ::L`: base loss, can either take (y, y_true) or (y, t_true) as input
+- `Ω::R`: regularization, takes y as input
+- `α::Float64`: default value of 1.0
+"""
+struct ImitationLoss{P,L,R}
+    maximizer::P
+    ℓ::L
+    Ω::R
+    α::Float64
+end
+
+function Base.show(io::IO, l::ImitationLoss)
+    (; ℓ, Ω, maximizer, α) = l
+    return print(io, "ImitationLoss($maximizer, $ℓ, $Ω, $α)")
+end
+
+"""
+    ImitationLoss(maximizer[; ℓ=(y,t_true)->0.0, Ω=y->0.0, α=1.0])
+
+Shorter constructor with defaults.
+"""
+function ImitationLoss(
+    maximizer; ℓ=(y, t_true) -> 0.0, base_loss=nothing, Ω=y -> 0.0, omega=nothing, α=1.0
```

Comment:
On second thought I'm not sure this habit of supporting both unicode and explicit names is good. I did it partly because LaTeX made it harder to show Julia snippets with unicode. What do you think?

Reply:
For instance here we don't even support it for alpha

Reply:
I would remove all explicit names support.
I would however rename field $\ell$ to `base_loss` in order to stay consistent with the `SSVMLoss` and avoid confusions between the loss `l` and its base loss $\ell$.

Reply:
okay let's do it

Reply:
I'd say a separate one